lobsterpy/structuregraph/graph.py

```diff
@@ -0,0 +1,171 @@
+
+        self.path_to_poscar = path_to_poscar
+        self.path_to_charge = path_to_charge
+        self.path_to_cohpcar = path_to_cohpcar
+        self.path_to_icohplist = path_to_icohplist
+        self.path_to_madelung = path_to_madelung
+        self.which_bonds = which_bonds
+        self.cutoff_icohp = cutoff_icohp
+
+        if self.which_bonds == "all":
+            self.additional_condition = 0
+        elif self.which_bonds == "cation-anion":
+            self.additional_condition = 1
+        else:
+            raise ValueError(
+                "Only accepted values are 'all' and 'cation-anion'."
+                "Please check the input parameters of which_bonds arg"
+            )
+        self.start = start
+
+        self.sg = self.get_decorated_sg()
+
+    def get_decorated_sg(self):
+        """
+        Method to generate graph object decorated with bonding data from lobsterpy
+        Returns:
+            structure graph object
+        """
+        if self.add_additional_data_sg:
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                perc_strength_ICOHP=self.cutoff_icohp,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=self.additional_condition,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+                filename_blist_sg1=self.path_to_icobilist,
+                id_blist_sg1="ICOBI",
+                filename_blist_sg2=self.path_to_icooplist,
+                id_blist_sg2="ICOOP",
+                valences_from_charges=True,
+                adapt_extremum_to_add_cond=True,
+            )
+
+        else:
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                perc_strength_ICOHP=self.cutoff_icohp,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=self.additional_condition,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+                valences_from_charges=True,
+                adapt_extremum_to_add_cond=True,
+            )
+
+        # Adds Mulliken and Löwdin charges as site properties to structure object (node properties)
+        decorated_structure = Charge(self.path_to_charge).get_structure_with_charges(
+            self.path_to_poscar
+        )
+
+        # Create the structure graph object decorated with site and edge properties based on ICOHP/ICOBI/ICOOP data
+        lobster_env = chemenvlobster.get_bonded_structure(
+            structure=decorated_structure, decorate=True, edge_properties=True
+        )
+
+        # Initialize automating bonding analysis from Lobsterpy based on ICOHP
+        analyze = Analysis(
```

Comment:
Isn't LobsterNeighbors part of the Analysis object? Can't you reuse this? I think we are currently running the same code multiple times which could be really slow.

Reply:
Indeed `LobsterNeighbors` is part of Analysis object which can be reused, but as it is not initialized with ICOOP and ICOBI info when we create the analyze object. So the structure graph will not have ICOBI and ICOOP as edge properties.

I think to keep it simple we can simply not have them in single structure graph object?

It was implemented like this as ICOOP and ICOBI extension was not implemented before. Now we can simply have different sg objects for based on different analysis objects. 

Let me know what you think of this? I will then make changes there

Reply:
Can't you add a kwarg or something to Analysis to pass the ICOBI, ICOOP stuff?

Reply:
I can, but I think it can be quite confusing if we are already planning to extend the analysis to COBIs and COOPs.

If the user does analysis for COBICAR, then sg will not have COHP info added with our current implementation in pymatgen. I will need to first update LobsterNeighbors if we want to go this way.

Reply:
Ideally, we can determine bonds based on ICOHP, ICOBI and add ICOHP, ICOBI values to each structure graph. I guess we will keep the current implementation for now and then think about improvements later.

Reply:
Thanks for checking!

Reply:
> Ideally, we can determine bonds based on ICOHP, ICOBI and add ICOHP, ICOBI values to each structure graph. I guess we will keep the current implementation for now and then think about improvements later.

Yes, I agree 😄

Reply:
> Thanks for checking!

You are welcome